scripts/run_forge.sh

```diff
@@ -0,0 +1,164 @@
+    FORGE_POD_NAME=$FORGE_NAMESPACE
+    kubectl delete pod $FORGE_POD_NAME || true
+    kubectl wait --for=delete "pod/${FORGE_POD_NAME}" || true
+
+    kubectl run $FORGE_POD_NAME \
+        --overrides='{ "spec": { "serviceAccount": "forge" }  }' \
+        --restart=Never \
+        --image="${AWS_ACCOUNT_NUM}.dkr.ecr.${AWS_REGION}.amazonaws.com/aptos/forge:$IMAGE_TAG" \
+        --command -- bash -c "forge test k8s-swarm --image-tag $IMAGE_TAG --namespace $FORGE_NAMESPACE $KEEP_ARGS $ENABLE_HAPROXY_ARGS"
+
+    # wait for enough time for the pod to start and potentially new nodes to come online
+    kubectl wait --timeout=5m --for=condition=Ready "pod/${FORGE_POD_NAME}"
+
+    # tail the logs and tee them for further parsing
+    kubectl logs -f $FORGE_POD_NAME | tee $FORGE_OUTPUT
+
+    # parse the pod status: https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-phase
+    forge_pod_status=$(kubectl get pod $FORGE_POD_NAME -o jsonpath="{.status.phase}")
+    echo "Forge pod status: ${forge_pod_status}"
+    if [ "$forge_pod_status" = "Succeeded" ]; then
+        FORGE_EXIT_CODE=0
+    else
+        FORGE_EXIT_CODE=1
+    fi
+fi
+
+FORGE_END_TIME_MS="$(date '+%s')000"
+
+# parse the JSON report
+cat $FORGE_OUTPUT | awk '/====json-report-begin===/{f=1;next} /====json-report-end===/{f=0} f' >"${FORGE_REPORT}"
+# If no report was generated, fill with default report
+if [ ! -s "${FORGE_REPORT}" ]; then
+    echo '{"text": "Forge test runner terminated"}' >"${FORGE_REPORT}"
+fi
+
+# detect regressions
+# TODO(rustielin): do not block on perf regressions for now until Forge network performance stabilizes
+avg_tps=$(cat $FORGE_REPORT | grep -oE '\d+ TPS' | awk '{print $1}')
+p99_latency=$(cat $FORGE_REPORT | grep -oE '\d+ ms p99 latency' | awk '{print $1}')
+if [ -n "$avg_tps" ] && [ "$avg_tps" -lt "$TPS_THRESHOLD" ]; then
+    echo "(\!) AVG_TPS: ${avg_tps} < ${TPS_THRESHOLD} tps"
+    # FORGE_EXIT_CODE=1
+fi
+if [ -n "$p99_latency" ] && [ "$p99_latency" -gt "$P99_LATENCY_MS_THRESHOLD" ]; then
+    echo "(\!) P99_LATENCY: ${p99_latency} > 5000 ms"
+    # FORGE_EXIT_CODE=1
+fi
+
+# If no report text was generated, fill with default text
+FORGE_REPORT_TXT=$(cat $FORGE_REPORT | jq -r .text)
+if [ -z "$FORGE_REPORT_TXT" ]; then
+    FORGE_REPORT_TXT="Forge report text empty. See test runner output."
+    FORGE_EXIT_CODE=1
+fi
+
+if [ "$FORGE_EXIT_CODE" = "0" ]; then
+    FORGE_COMMENT_HEADER='### :white_check_mark: Forge test success'
+else
+    FORGE_COMMENT_HEADER='### :x: Forge test failure'
+fi
+
+# remove the "aptos-" prefix to get the chain name as reported to Prometheus
+FORGE_CHAIN_NAME=${FORGE_CLUSTER_NAME#"aptos-"}
+FORGE_DASHBOARD_LINK="${GRAFANA_BASE_URL}&var-namespace=${FORGE_NAMESPACE}&var-chain_name=${FORGE_CHAIN_NAME}&from=${FORGE_START_TIME_MS}&to=${FORGE_END_TIME_MS}"
+if [ -z "$GRAFANA_BASE_URL" ]; then
+    echo "GRAFANA_BASE_URL not set. Use above query on supported Grafana dashboards"
```

Comment:
considering the amount of non-trivial grepping awking and jqing you're doing in this script, I think mid term it's better to write these things in a real programming language like Rust or TypeScript just fyi, but no need to block this PR. Still better than having the script inline in the GHA job for sure.

Reply:
I mean generally speaking those are solvable problems imo. The build overhead is probably not that high anymore if you create a separate create since we remove workspace-hack, so if you do cargo run on a small crate that has few dependencies, it shouldn't take very long nowadays, even with an empty cache.
This is especially the case if you keep the crate small/focused and don't use a ton of dependencies which you probably don't need for something like what this script here does.
There is also things like https://github.com/fornwall/rust-script that reduces a bit the overhead of creating many Cargo.tomls, but I'm not sure how well this would be understood by an IDE.
Generally speaking Rust's compiler is pretty fast for scoped crates so the build overhead isn't a big deal imo.

As for why it gets more confusing with forge-test-manager / forge-cli I didn't really understand the concern? Is this just a naming issue?